syntax.Rmd

```diff
@@ -465,26 +465,49 @@ do_something_very_complicated("that", requires, many, arguments,
 
 As described under [Named arguments](#argument-names), you can omit the argument names
 for very common arguments (i.e. for arguments that are used in almost every
-invocation of the function). Short unnamed arguments can also go on the same
-line as the function name, even if the whole function call spans multiple lines.
+invocation of the function). If this introduces a large disparity between the line lengths, you may want to supply names anyway:
 
 ```{r, eval = FALSE}
-map(x, f,
+# Good
+map(
+  x,
+  f,
+  extra_argument_a = 10,
+  extra_argument_b = c(1, 43, 390, 210209)
+)
+
+# Also good
+map(
+  x = x,
+  f = f,
   extra_argument_a = 10,
   extra_argument_b = c(1, 43, 390, 210209)
 )
 ```
 
-You may also place several arguments on the same line if they are closely
-related to each other, e.g., strings in calls to `paste()` or `stop()`. When
-building strings, where possible match one line of code to one line of output.
+You may also place multiple unnamed arguments on the same line if they are closely
+related to each other, e.g., strings in calls to `paste()` or arguments to `sprintf()`. When
+building strings, match one line of code to one line of output where possible.
 
 ```{r, eval = FALSE}
 # Good
 paste0(
   "Requirement: ", requires, "\n",
   "Result: ", result, "\n"
 )
```

Comment:
This one is likely quite hard for a formatter to do. It's probably either:
- "put it all on one line if it fits"
- "put each arg on its own line"

Reply:
I suspect an LLM-assisted formatter could do a decent job :)

Reply:
Yeah, I think this is a case where a formatter might just leave unwrapped lines alone. But maybe only for a handful of white-listed functions?